### DISCUSSION TITLE
Sizing: rem vs. em. vs px

### DISCUSSION THREAD

**Discussion by stratofax** [cadentcom/pico-wp-block-theme#25] [Ideas]
Unless there's a specific reason to do otherwise, should we use `rem` as the default unit for all sizing and layout dimensions?

This includes:

- font sizes
- layout dimensions (size, padding, etc.)

In favor: [When and Where to Use REM and EM Units in Web Design? | HackerNoon](https://hackernoon.com/when-and-where-to-use-rem-and-em-units-in-web-design)

**Comment by stratofax** [comment]
@ratulkhan74 Any thoughts on using `rem` universally throughout the layout?

**Reply by stratofax** [reply]
I imagine that some of the WordPress blocks use only `px` and don't support `rem` units.

**Reply by ratulkhan74** [reply]
@stratofax `px` is the universally used default unit system for all kind of sizing and layout dimensions `px` unit system more easy to handle than `rem` or `em` i personally use `px` for all kind of sizing. I can change our pico theme unit system `px` to `rem` from the `theme.json` file. I will change though. By the way wordpress blocks show  `px` as default unit system.

**Reply by stratofax** [reply]
@ratulkhan74 Since `rem` is specifically intended to improve accessibility by maintaining the relative sizes of elements, regardless of the physical size or resolution of the viewport, it also users to magnify or shrink the display size without breaking the layout. 

As one of the specific goals of the pico-block theme is to maximize accessibility, we'll use `rem` in all cases, unless there's a specific need for `em` or `px`. One situation where I imagine `px` might be a better choice is defining the breakpoints for a set of responsive layouts in the CSS files. Otherwise, let's use `rem`. 

Please discuss possible and specific uses of `px` or `em` here before implementing these units in the code.

**Reply by ratulkhan74** [reply]
@stratofax I already changed `px` to `rem` but block theme block elements by default scal with `px`

**Reply by stratofax** [reply]
@ratulkhan74 I thought the WP blocks would limit us to `px` and that's fine. We can also specify `rem` sizes in `theme.json` and the CSS file by defining our own classes, and assign a `class` to the WP block element.

Of course the user can override this by setting the `px` sizes in the block, but we don't need or want to do that if we can set all the sizes in `theme.json` using classes.
